go/fn/run.go

```diff
@@ -58,10 +58,10 @@ func Run(p ResourceListProcessor, input []byte) (out []byte, err error) {
 		v := recover()
 		if v != nil {
 			switch t := v.(type) {
-			case *ErrKubeObjectFields:
-				err = t
-			case *ErrSubObjectFields:
-				err = t
+			case ErrKubeObjectFields:
```

Comment:
To be more robust, I'd suggest doing type switch on both pointer and non-pointer types.

Reply:
Done.